plugins/modules/proxmox.py

```diff
@@ -518,22 +756,29 @@ def update_config(self, vmid, node, disk, cpus, memory, swap, **kwargs):
                 )
 
         # Remove all empty kwarg entries
-        kwargs = dict((k, v) for k, v in kwargs.items() if v is not None)
+        kwargs = dict((key, val) for key, val in kwargs.items() if val is not None)
 
         if cpus is not None:
             kwargs["cpulimit"] = cpus
         if disk is not None:
-            kwargs["rootfs"] = disk
+            kwargs["disk_volume"] = parse_disk_string(disk)
+        if "disk_volume" in kwargs:
+            disk_dict = build_volume(key="rootfs", **kwargs.pop("disk_volume"))
```

Comment:
Here you should make sure that the values in `disk_volume` are all strings.

It's probably best to convert them to a string (and maybe emit a warning if they aren't).